tests/unit/test_utils.py

```diff
@@ -146,6 +146,14 @@ def test_open_guess_encoding():
         ("git+https://github.com/jupyterhub/repo2docker", False),
         ("numpy", False),
         ("# -e .", False),
+        ("--pre", False),
+        # pip ignores the package name and treats this like `--pre` on a line
+        # by itself
+        ("--pre pandas", False),
+        # These are invalid lines as far as pip is concerned, check that our
+        # code is robust and continues running
+        ("--unrecognized", True),
```

Comment:
should return False, I think? Or do we want to make any unhandled flag result in the less optimized behavior?